registry-v2.md

```diff
@@ -0,0 +1,82 @@
+# Registry v2
+
+*NOTE: This registry is not yet deployed on hex.pm*
+
+The first version of the registry format used a single file for the entire registry. While this was efficient in the number of HTTP requests needed to resolve dependencies, the size of the registry would grow out of control as the repository grows.
+
+The new version of the registry format is split into multiple files to make it scale as the repository grows with more packages.
+
+For every package in the repository there is a package registry file under `/packages/[NAME]` in the repository, it contains all the releases of that package and all dependencies of the releases. There are also two index files `/names` and `/versions` that contains all package names and all package names+versions respectively.
+
```

Comment:
The way I've been thinking about it (but haven't discussed it with Tristan) is going in and eventually keeping the same interface we have today with the caveat that we can't pre-fetch an index anymore under a new API.

What we'd do is cache and maintain local copies of apps that have been fetched before into a partial index that we can populate and fill as we go. A rebar3 update would reupdate the state of all known apps. I'm not quite sure what we would do for `rebar3 deps` though, which currently lists dependencies for a living.